test/discrete/test_pg.py

```diff
@@ -79,6 +80,7 @@ def test_pg(args=get_args()):
         optim,
         dist,
         args.gamma,
+        action_scaling=isinstance(env.action_space, Box),
```

Comment:
See last part of the PR description:

> I added input validation for things like state and action_scaling (which only makes sense for continuous envs). After adding this, some tests failed to pass this validation. There I added action_scaling=isinstance(env.action_space, Box), after which tests were green. I don't know why the tests were green before, since action scaling doesn't make sense for discrete actions. I guess some aspect was not tested and didn't crash.

Reply:
hmm interesting... and good catch!